indico/web/util.py

```diff
@@ -179,3 +181,34 @@ def get_request_info(hide_passwords=True):
 def url_for_index(_external=False, _anchor=None):
     from indico.web.flask.util import url_for
     return url_for('categories.display', _external=_external, _anchor=_anchor)
+
+
+def signed_url_for(user, blueprint, url_params=None, *args, **kwargs):
+    from indico.web.flask.util import url_for
+    _external = kwargs.pop('_external', False)
+    base_url = url_for(blueprint, *args, **(url_params or {}))
+    qs = url_encode(sorted(kwargs.items()))
+    url = '{}?{}'.format(base_url, qs) if qs else base_url
+    signer = Signer(user.signing_secret, salt='url-signing')
+    print 'signing', url
+    qs = url_encode(dict(kwargs, token=signer.get_signature(url)))
+    full_base_url = url_for(blueprint, *args, _external=_external, **(url_params or {}))
+    return '{}?{}'.format(full_base_url, qs) if qs else base_url
+
+
+def is_signed_url_valid(user, url):
+    parsed = url_parse(url)
+    params = url_decode(parsed.query)
+    if 'token' not in params:
+        return False
+    signature = params.pop('token')
+    url = url_unparse((
+        '',
+        '',
+        parsed.path,
+        url_encode(sorted(params.items()), sort=True),
+        parsed.fragment
+    ))
+    signer = Signer(user.signing_secret, salt='url-signing')
+    print url, signature, signer.verify_signature(url, signature)
```

Comment:
Forgot to remove? :P

Reply:
🤦‍♂

Reply:
It was actually added on a later fixup, because before opening a PR I grepped for `print` and `console` :D